File: src/main/java/org/cubexmc/railway/service/OperationMode.java
package org.cubexmc.railway.service;

public enum OperationMode {
    LOCAL,
    GLOBAL;

    public static OperationMode from(String s, OperationMode def) {
        if (s == null) return def;
        switch (s.trim().toLowerCase()) {
            case "local": return LOCAL;
            case "global": return GLOBAL;
            default: return def;
        }
    }
}

File: src/main/java/org/cubexmc/metro/service/OperationMode.kt
package org.cubexmc.metro.service

import java.util.Locale

enum class OperationMode {
    LOCAL,
    GLOBAL,
    ;

    companion object {
        @JvmStatic
        fun from(s: String?, def: OperationMode?): OperationMode? {
            if (s == null) return def
            return when (s.trim().lowercase(Locale.getDefault())) {
                "local" -> LOCAL
                "global" -> GLOBAL
                else -> def
            }
        }
    }
}
